config/versions/v0.go

```diff
@@ -0,0 +1,24 @@
+package versions
+
+import (
+	"context"
+)
+
+// Version0 is a baseline version with no changes, so we can downgrade back to nothing
+// It does not implement any upgrade interfaces
+type Version0 struct {
+}
+
+func init() {
+	Manager.registerVersion(0, &Version0{})
+}
```

Comment:
`init` are read in filename order. You have limited this functionality to a total of 10 versions using your current naming convention

I created [10 upgrade version files ](https://github.com/user-attachments/assets/1d55d813-49d6-4445-a771-729bc5244b75)and structs for testing, in each I add a println to highlight this:

```go
func init() {
	fmt.Println("v0")
	Manager.registerVersion(0, &Version0{})
}
```

```
 go run . upgrade
v0
v1
v10
v2
v3
v4
v5
v6
v7
v8
v9
GoCryptoTrader: config-helper tool
version registered out of sequence: 10
exit status 2
```

Easiest solution while still wanting to rely on `init` is to rename files to have leading zeroes.

Reply:
Pretty sure that I can revamp that sequence requirement, since I added version number in.
And I don't think it's strictly necessary.

Reply:
Fixed [d4eb81025](https://github.com/gbjk/gocryptotrader/commit/d4eb81025)